plugin.image.qwantz/plugin.py

```diff
@@ -0,0 +1,165 @@
+
+import sys
+from bs4 import BeautifulSoup
+import os
+import random
+from traceback import format_exc
+import urllib
+import urllib2
+import urlparse
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+
+# Constants
+ADDON = xbmcaddon.Addon()
+LATEST_FOLDER_NAME = ADDON.getLocalizedString(32001)
+RANDOM_FOLDER_NAME = ADDON.getLocalizedString(32002)
+PREV_FOLDER_NAME = ADDON.getLocalizedString(32003)
+NEXT_FOLDER_NAME = ADDON.getLocalizedString(32004)
+BROWSE_FOLDER_NAME = ADDON.getLocalizedString(32005)
+BROWSE_RANGE_FOLDER_NAME = ADDON.getLocalizedString(32006)
+QWANTZ_URL = 'http://www.qwantz.com/'
+
+def build_url(query):
+    """ Builds the URL for Kodi to traverse pages """
+    return base_url + '?' + urllib.urlencode(query)
+
+def get_latest_comic_id():
+    """ Returns the ID of the most recently published comic """
+    try:
+        soup = BeautifulSoup(urllib2.urlopen(QWANTZ_URL), "html.parser")
+        prevId = soup.find_all('a', { 'title': 'Previous comic' })[0]['href'].split('comic=')[1]
+        curId = int(prevId) + 1
+        xbmc.log('curId is ' + str(curId))
+        return curId
+    except Exception:
+        xbmc.log(format_exc())
+        return 1 # default to first comic if we can't grab latest
+
+def add_comic_item(url, includePrevAndNext):
+    """ Parses the HTML at the specified URL to grab the comic's information and add it to the results """
+    try:
+        xbmc.log('url is ' + url)
+        soup = BeautifulSoup(urllib2.urlopen(url), "html.parser")
+        title = soup.title.string
+        date = title.split(' - ')[1]
+        img = soup.find_all('img', { 'class': 'comic' })[0]
+        imgSrc = QWANTZ_URL + img['src']
+        altText = img['title']
+        
+        li = xbmcgui.ListItem(date, iconImage='DefaultPicture.png')
+        li.setProperty('description', altText)
+        xbmc.log('setting img with addon_handle ' + str(addon_handle))
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=imgSrc, listitem=li)
+        
+        if includePrevAndNext:
+            add_arrow_item(soup, 'Previous comic', PREV_FOLDER_NAME, 'leftarrow.png')
+            add_arrow_item(soup, 'Next comic', NEXT_FOLDER_NAME, 'rightarrow.png')
+    except Exception:
+        xbmc.log(format_exc())
+    
+    
+            
+def add_arrow_item(soup, titleText, folderName, iconName):
+    """ Adds the previous or next arrow button for the comic.
+        The first comic won't have a previous arrow,
+        and the most recent comic won't have a next arrow.
+    """
+    try:
+        xbmc.log('Getting arrows')
+        btns = soup.find_all('a', { 'title': titleText })
+        if btns:
+            href = QWANTZ_URL + btns[0]['href'] # contains link to comic, e.g. http://www.qwantz.com/index.php?comic=3184
+            xbmc.log('href = ' + href)
+            iconPath = os.path.join(xbmcaddon.Addon().getAddonInfo('path'), 'resources', iconName)
```

Comment:
xbmcaddon.Addon.getAddonInfo("icon")

Reply:
I think this one is correct as is, because I want to load an image from the resources folder, and the image is not the same as the add-on icon.